### ISSUE TITLE
NVENC FFmpeg Encoder crashes while encoding

### ISSUE THREAD

**Issue by sungruhyun** [Xaymar/obs-StreamFX#940]
### Operating System

Windows 10 1903 and higher

### OBS Studio Version

28.0.1

### StreamFX Version

0.12.0 Alpha 134

### OBS Studio Log

<details>

```
13:15:12.030: CPU Name: Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz 
13:15:12.030: CPU Speed: 3492MHz 
13:15:12.030: Physical Cores: 4, Logical Cores: 8 
13:15:12.030: Physical Memory: 16308MB Total, 6436MB Free 
13:15:12.030: Windows Version: 10.0 Build 19043 (release: 2009; revision: 2006; 64-bit) 
13:15:12.030: Running as administrator: true 
13:15:12.030: Aero is Enabled (Aero is always on for windows 8 and above) 
13:15:12.031: Windows 10/11 Gaming Features: 
13:15:12.031: 	Game Bar: Off 
13:15:12.031: 	Game DVR: Off 
13:15:12.031: 	Game DVR Background Recording: Off 
13:15:12.031: 	Game Mode: On 
13:15:12.032: Sec. Software Status: 
13:15:12.033:  
13:15:12.033:  
13:15:12.034: Current Date/Time: 2022-09-22, 13:15:12 
13:15:12.034: Browser Hardware Acceleration: true 
13:15:12.034: Hide OBS windows from screen capture: false 
13:15:12.034: Qt Version: 6.3.1 (runtime), 6.3.1 (compiled) 
13:15:12.034: Portable mode: false 
13:15:12.688: OBS 28.0.1 (64-bit, windows) 
13:15:12.688: --------------------------------- 
13:15:12.689: --------------------------------- 
13:15:12.689: audio settings reset: 
13:15:12.689: 	samples per sec: 48000 
13:15:12.689: 	speakers:        2 
13:15:12.689: 	max buffering:   960 milliseconds 
13:15:12.689: 	buffering type:  dynamically increasing 
13:15:12.690: --------------------------------- 
13:15:12.690: Initializing D3D11... 
13:15:12.690: Available Video Adapters:  
13:15:12.692: 	Adapter 0: NVIDIA GeForce GTX 960 
13:15:12.692: 	  Dedicated VRAM: 2105212928 
13:15:12.692: 	  Shared VRAM:    4255324160 
13:15:12.692: 	  PCI ID:         10de:1401 
13:15:12.693: 	  Driver Version: 30.0.15.1215 
13:15:12.693: 	  output 0: 
13:15:12.693: 	    name=C32JG5x 
13:15:12.693: 	    pos={0, 0} 
13:15:12.693: 	    size={1920, 1080} 
13:15:12.693: 	    attached=true 
13:15:12.693: 	    refresh=144 
13:15:12.693: 	    space=RGB_FULL_G22_NONE_P709 
13:15:12.693: 	    sdr_white_nits=80 
13:15:12.693: 	    nit_range=[min=0.500000, max=270.000000, max_full_frame=270.000000] 
13:15:12.693: 	Adapter 1: Intel(R) HD Graphics 4600 
13:15:12.693: 	  Dedicated VRAM: 117964800 
13:15:12.693: 	  Shared VRAM:    2147483648 
13:15:12.693: 	  PCI ID:         8086:412 
13:15:12.694: 	  Driver Version: 20.19.15.5171 
13:15:12.694: 	Adapter 2: NVIDIA GeForce GTX 960 
13:15:12.694: 	  Dedicated VRAM: 2105212928 
13:15:12.694: 	  Shared VRAM:    4255324160 
13:15:12.694: 	  PCI ID:         10de:1401 
13:15:12.694: 	  Driver Version: 30.0.15.1215 
13:15:12.696: Loading up D3D11 on adapter NVIDIA GeForce GTX 960 (0) 
13:15:12.766: D3D11 loaded successfully, feature level used: b000 
13:15:12.766: DXGI increase maximum frame latency success 
13:15:12.766: D3D11 GPU priority setup success 
13:15:15.011: --------------------------------- 
13:15:15.011: video settings reset: 
13:15:15.011: 	base resolution:   1920x1080 
13:15:15.011: 	output resolution: 1920x1080 
13:15:15.011: 	downscale filter:  Lanczos 
13:15:15.011: 	fps:               60/1 
13:15:15.011: 	format:            NV12 
13:15:15.011: 	YUV mode:          Rec. 709/Full 
13:15:15.011: NV12 texture support enabled 
13:15:15.011: P010 texture support not available 
13:15:15.013: Audio monitoring device: 
13:15:15.013: 	name: 기본값 
13:15:15.013: 	id: default 
13:15:15.014: --------------------------------- 
13:15:15.016: No AJA devices found, skipping loading AJA UI plugin 
13:15:15.016: Failed to initialize module 'aja-output-ui.dll' 
13:15:15.019: No AJA devices found, skipping loading AJA plugin 
13:15:15.019: Failed to initialize module 'aja.dll' 
13:15:15.019: Skipping module '../../obs-plugins/64bit/chrome_elf.dll', not an OBS plugin 
13:15:15.027: [CoreAudio encoder]: Adding CoreAudio AAC encoder 
13:15:15.028: Failed to load 'en-US' text for module: 'decklink-captions.dll' 
13:15:15.029: Failed to load 'en-US' text for module: 'decklink-output-ui.dll' 
13:15:15.033: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed 
13:15:15.033: Failed to initialize module 'decklink.dll' 
13:15:15.060: [AMF] Unable to load 'amfrt64.dll', error code 126. 
13:15:15.062: [AMF] AMF Test failed due to one or more errors. 
13:15:15.062: Failed to initialize module 'enc-amf.dll' 
13:15:15.088: Skipping module '../../obs-plugins/64bit/libcef.dll', not an OBS plugin 
13:15:15.088: Skipping module '../../obs-plugins/64bit/libEGL.dll', not an OBS plugin 
13:15:15.088: Skipping module '../../obs-plugins/64bit/libGLESv2.dll', not an OBS plugin 
13:15:15.097: [obs-browser]: Version 2.18.5 
13:15:15.097: [obs-browser]: CEF Version 103.0.5060.134 (runtime), 103.0.0-5060-shared-textures.2591+g4204d54+chromium-103.0.5060.134 (compiled) 
13:15:15.101: NVENC supported 
13:15:15.104: [noise suppress]: NVIDIA RTX denoiser disabled, redistributable not found 
13:15:15.104: [NVIDIA RTX VIDEO FX]: FX disabled, redistributable not found. 
13:15:15.143: [obs-websocket] [obs_module_load] you can haz websockets (Version: 5.0.1 | RPC Version: 1) 
13:15:15.144: [obs-websocket] [obs_module_load] Qt version (compile-time): 6.3.1 | Qt version (run-time): 6.3.1 
13:15:15.144: [obs-websocket] [obs_module_load] Linked ASIO Version: 101201 
13:15:15.600: [obs-websocket] [obs_module_load] Module loaded. 
13:15:15.612: [vlc-video]: Couldn't find VLC installation, VLC video source disabled 
13:15:15.625: [StreamFX] Loading Version 0.12.0a134-g6853cc6a 
13:15:15.678: [StreamFX] <gfx::opengl> Version 0.0 initialized. 
13:15:15.682: [StreamFX] <nvidia::cuda::cuda> Driver reported CUDA version: 11.6.0 
13:15:15.748: [StreamFX] <nvidia::cuda::context> Initialized CUDA on device 'NVIDIA GeForce GTX 960' (8ba0ed93-06b9-acb3-492f-9e2ef08b7699, 0000c913, 1). 
13:15:15.755: [StreamFX] <nvidia::cv::cv> No supported NVIDIA SDK is installed to provide 'NVCVImage.dll'. 
13:15:15.755: [StreamFX] <filter::autoframing> Failed to make NVIDIA providers available due to error: Failed to load 'NVCVImage.dll'. 
13:15:15.755: [StreamFX] <filter::autoframing> All supported providers failed to initialize, disabling effect. 
13:15:15.755: [StreamFX] <nvidia::cv::cv> No supported NVIDIA SDK is installed to provide 'NVCVImage.dll'. 
13:15:15.755: [StreamFX] <filter::video_denoising> Failed to make NVIDIA providers available due to error: Failed to load 'NVCVImage.dll'. 
13:15:15.755: [StreamFX] <filter::video_denoising> All supported providers failed to initialize, disabling effect. 
13:15:15.755: [StreamFX] <nvidia::cv::cv> No supported NVIDIA SDK is installed to provide 'NVCVImage.dll'. 
13:15:15.755: [StreamFX] <filter::video_superresolution> Failed to make NVIDIA Super-Resolution available due to error: Failed to load 'NVCVImage.dll'. 
13:15:15.755: [StreamFX] <filter::video_superresolution> All supported Super-Resolution providers failed to initialize, disabling effect. 
13:15:15.755: [StreamFX] <nvidia::cv::cv> No supported NVIDIA SDK is installed to provide 'NVCVImage.dll'. 
13:15:15.756: [StreamFX] <filter::virtual_greenscreen> Failed to make NVIDIA Greenscreen available due to error: Failed to load 'NVCVImage.dll'. 
13:15:15.756: [StreamFX] <filter::virtual_greenscreen> All supported Virtual Greenscreen providers failed to initialize, disabling effect. 
13:15:15.756: [StreamFX] Loaded Version 0.12.0a134-g6853cc6a 
13:15:15.791: --------------------------------- 
13:15:15.791:   Loaded Modules: 
13:15:15.791:     StreamFX.dll 
13:15:15.791:     win-wasapi.dll 
13:15:15.791:     win-mf.dll 
13:15:15.791:     win-dshow.dll 
13:15:15.791:     win-capture.dll 
13:15:15.791:     vlc-video.dll 
13:15:15.791:     text-freetype2.dll 
13:15:15.791:     rtmp-services.dll 
13:15:15.791:     obs-x264.dll 
13:15:15.791:     obs-websocket.dll 
13:15:15.791:     obs-vst.dll 
13:15:15.791:     obs-transitions.dll 
13:15:15.791:     obs-text.dll 
13:15:15.791:     obs-qsv11.dll 
13:15:15.791:     obs-outputs.dll 
13:15:15.791:     obs-filters.dll 
13:15:15.791:     obs-ffmpeg.dll 
13:15:15.791:     obs-browser.dll 
13:15:15.791:     image-source.dll 
13:15:15.791:     frontend-tools.dll 
13:15:15.791:     decklink-output-ui.dll 
13:15:15.791:     decklink-captions.dll 
13:15:15.791:     coreaudio-encoder.dll 
13:15:15.791: --------------------------------- 
13:15:15.792: ==== Startup complete =============================================== 
13:15:15.954: All scene data cleared 
13:15:15.954: ------------------------------------------------ 
13:15:15.990: [win-wasapi: '데스크탑 오디오'] update settings: 
13:15:15.990: 	device id: default 
13:15:15.990: 	use device timing: 1 
13:15:15.997: User selected new VST plugin: 'C:/Program Files/VSTPlugins/ReaPlugs/reafir_standalone.dll' 
13:15:16.002: [Loaded global audio device]: '데스크탑 오디오' 
13:15:16.002:     - filter: 'VST 2.x 플러그인' (vst_filter) 
13:15:16.004: [win-wasapi: '마이크/보조'] update settings: 
13:15:16.004: 	device id: default 
13:15:16.004: 	use device timing: 0 
13:15:16.004: [Loaded global audio device]: '마이크/보조' 
13:15:16.011: [duplicator-monitor-capture: '디스플레이 캡쳐'] update settings: 
13:15:16.011: 	display: 1 (1920x1080) 
13:15:16.011: 	cursor: true 
13:15:16.011: 	method: DXGI 
13:15:16.011: Switched to scene '장면' 
13:15:16.012: ------------------------------------------------ 
13:15:16.012: Loaded scenes: 
13:15:16.012: - scene '장면': 
13:15:16.012:     - source: '디스플레이 캡쳐' (monitor_capture) 
13:15:16.012: ------------------------------------------------ 
13:15:16.029: WASAPI: Device 'VoiceMeeter Output(VB-Audio VoiceMeeter VAIO)' [48000 Hz] initialized 
13:15:16.030: WASAPI: Device 'VoiceMeeter Input(VB-Audio VoiceMeeter VAIO)' [48000 Hz] initialized 
13:15:19.875: [StreamFX] [h264_nvenc] Configuration: 
13:15:19.875: [StreamFX] [h264_nvenc]   FFmpeg: 
13:15:19.875: [StreamFX] [h264_nvenc]     Custom Settings:  
13:15:19.875: [StreamFX] [h264_nvenc]     Standard Compliance: Normal 
13:15:19.875: [StreamFX] [h264_nvenc]     Threading: Slice & Frame (with 1 threads) 
13:15:19.875: [StreamFX] [h264_nvenc]   Video: 
13:15:19.875: [StreamFX] [h264_nvenc]     Texture: 1920x1080 nv12 BT.709 pc 
13:15:19.875: [StreamFX] [h264_nvenc]     Framerate: 60/2 (30.000000 FPS) 
13:15:19.875: [StreamFX] [h264_nvenc]   Keyframes:  
13:15:19.875: [StreamFX] [h264_nvenc]     Distance: 60 frames 
13:15:19.875: [StreamFX] [h264_nvenc]   NVIDIA NVENC: 
13:15:19.875: [StreamFX] [h264_nvenc]     Preset: p7 
13:15:19.875: [StreamFX] [h264_nvenc]     Rate Control: constqp 
13:15:19.875: [StreamFX] [h264_nvenc]       Two Pass: Disabled 
13:15:19.875: [StreamFX] [h264_nvenc]       Multi-Pass: fullres 
13:15:19.875: [StreamFX] [h264_nvenc]       Look-Ahead: 32 Frames 
13:15:19.875: [StreamFX] [h264_nvenc]       Adaptive I-Frames: Enabled <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]       Adaptive B-Frames: Enabled <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]       Bitrate: 
13:15:19.875: [StreamFX] [h264_nvenc]         Target: 0 bits/sec 
13:15:19.875: [StreamFX] [h264_nvenc]         Minimum: 0 bits/sec <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]         Maximum: 0 bits/sec <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]         Buffer: 0 bits <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]       Quality: 
13:15:19.875: [StreamFX] [h264_nvenc]         Target: 0  <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]         Minimum: -1  
13:15:19.875: [StreamFX] [h264_nvenc]         Maximum: -1  
13:15:19.875: [StreamFX] [h264_nvenc]       Quantization Parameters: 
13:15:19.875: [StreamFX] [h264_nvenc]         I-Frame: 34  
13:15:19.875: [StreamFX] [h264_nvenc]         P-Frame: 34  
13:15:19.875: [StreamFX] [h264_nvenc]         B-Frame: 34  
13:15:19.875: [StreamFX] [h264_nvenc]         CB Offset: 0  <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]         CR Offset: 0  <Default> 
13:15:19.875: [StreamFX] [h264_nvenc]     B-Frames: 4 Frames 
13:15:19.875: [StreamFX] [h264_nvenc]       Reference Mode: each 
13:15:19.875: [StreamFX] [h264_nvenc]     Adaptive Quantization: 
13:15:19.875: [StreamFX] [h264_nvenc]       Spatial AQ: Enabled 
13:15:19.875: [StreamFX] [h264_nvenc]         Strength: 15  
13:15:19.876: [StreamFX] [h264_nvenc]       Temporal AQ: Enabled 
13:15:19.876: [StreamFX] [h264_nvenc]     Other: 
13:15:19.876: [StreamFX] [h264_nvenc]       Zero Latency: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Weighted Prediction: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Non-reference P-Frames: Enabled 
13:15:19.876: [StreamFX] [h264_nvenc]       Reference Frames: 0 Frames 
13:15:19.876: [StreamFX] [h264_nvenc]       Strict GOP: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Access Unit Delimiters: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Bluray Compatibility: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       A53 Closed Captions: Enabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       DPB Size: 0 Frames <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       DPB Size: 0 Frames <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Extra SEI Data: Enabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       User SEI Data: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Intra-Refresh: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Single Slice Intra-Refresh: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]       Constrained Encoding: Disabled <Default> 
13:15:19.876: [StreamFX] [h264_nvenc]     H.264/AVC: 
13:15:19.876: [StreamFX] [h264_nvenc]       Profile: high 
13:15:19.876: [StreamFX] [h264_nvenc]       Level: auto <Default> 
13:15:19.967: [CoreAudio AAC: 'Track1']: settings: 
13:15:19.967: 	mode:          AAC 
13:15:19.967: 	bitrate:       128 
13:15:19.967: 	sample rate:   48000 
13:15:19.967: 	cbr:           on 
13:15:19.967: 	output buffer: 1536 
13:15:19.980: ==== Recording Start =============================================== 
13:15:19.980: [ffmpeg muxer: 'adv_file_output'] Writing file 'F:/Users/sungruhyun/Videos/2022-09-22 13-15-19.mkv'... 
13:15:20.016: [StreamFX] [h264_nvenc] Configuration: 
13:15:20.016: [StreamFX] [h264_nvenc]   FFmpeg: 
13:15:20.016: [StreamFX] [h264_nvenc]     Custom Settings:  
13:15:20.016: [StreamFX] [h264_nvenc]     Standard Compliance: Normal 
13:15:20.016: [StreamFX] [h264_nvenc]     Threading: Slice & Frame (with 1 threads) 
13:15:20.016: [StreamFX] [h264_nvenc]   Video: 
13:15:20.016: [StreamFX] [h264_nvenc]     Texture: 1920x1080 nv12 BT.709 pc 
13:15:20.016: [StreamFX] [h264_nvenc]     Framerate: 60/2 (30.000000 FPS) 
13:15:20.016: [StreamFX] [h264_nvenc]   Keyframes:  
13:15:20.016: [StreamFX] [h264_nvenc]     Distance: 60 frames 
13:15:20.016: [StreamFX] [h264_nvenc]   NVIDIA NVENC: 
13:15:20.016: [StreamFX] [h264_nvenc]     Preset: p7 
13:15:20.016: [StreamFX] [h264_nvenc]     Rate Control: constqp 
13:15:20.016: [StreamFX] [h264_nvenc]       Two Pass: Disabled 
13:15:20.016: [StreamFX] [h264_nvenc]       Multi-Pass: fullres 
13:15:20.016: [StreamFX] [h264_nvenc]       Look-Ahead: 32 Frames 
13:15:20.016: [StreamFX] [h264_nvenc]       Adaptive I-Frames: Enabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Adaptive B-Frames: Enabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Bitrate: 
13:15:20.016: [StreamFX] [h264_nvenc]         Target: 0 bits/sec 
13:15:20.016: [StreamFX] [h264_nvenc]         Minimum: 0 bits/sec <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]         Maximum: 0 bits/sec <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]         Buffer: 0 bits <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Quality: 
13:15:20.016: [StreamFX] [h264_nvenc]         Target: 0  <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]         Minimum: -1  
13:15:20.016: [StreamFX] [h264_nvenc]         Maximum: -1  
13:15:20.016: [StreamFX] [h264_nvenc]       Quantization Parameters: 
13:15:20.016: [StreamFX] [h264_nvenc]         I-Frame: 34  
13:15:20.016: [StreamFX] [h264_nvenc]         P-Frame: 34  
13:15:20.016: [StreamFX] [h264_nvenc]         B-Frame: 34  
13:15:20.016: [StreamFX] [h264_nvenc]         CB Offset: 0  <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]         CR Offset: 0  <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]     B-Frames: 4 Frames 
13:15:20.016: [StreamFX] [h264_nvenc]       Reference Mode: each 
13:15:20.016: [StreamFX] [h264_nvenc]     Adaptive Quantization: 
13:15:20.016: [StreamFX] [h264_nvenc]       Spatial AQ: Enabled 
13:15:20.016: [StreamFX] [h264_nvenc]         Strength: 15  
13:15:20.016: [StreamFX] [h264_nvenc]       Temporal AQ: Enabled 
13:15:20.016: [StreamFX] [h264_nvenc]     Other: 
13:15:20.016: [StreamFX] [h264_nvenc]       Zero Latency: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Weighted Prediction: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Non-reference P-Frames: Enabled 
13:15:20.016: [StreamFX] [h264_nvenc]       Reference Frames: 0 Frames 
13:15:20.016: [StreamFX] [h264_nvenc]       Strict GOP: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Access Unit Delimiters: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Bluray Compatibility: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       A53 Closed Captions: Enabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       DPB Size: 0 Frames <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       DPB Size: 0 Frames <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Extra SEI Data: Enabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       User SEI Data: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Intra-Refresh: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Single Slice Intra-Refresh: Disabled <Default> 
13:15:20.016: [StreamFX] [h264_nvenc]       Constrained Encoding: Disabled <Default> 
13:15:20.017: [StreamFX] [h264_nvenc]     H.264/AVC: 
13:15:20.017: [StreamFX] [h264_nvenc]       Profile: high 
13:15:20.017: [StreamFX] [h264_nvenc]       Level: auto <Default>
```
</details>

### Current and Expected Behavior

OBS off

### Steps to Reproduce the Problem

With the encoder set to NVENC H264 (VIA FFMPEG), the color type is set to NV12, and the recording is turned on and touched OBS.

**Comment by Xaymar** [comment]
FYI the log field is to paste a log file as a file, or a link. Pasting it like this removes all formatting, which might even hide important information.

**Comment by Xaymar** [comment]
#### Confirmed
Was able to reproduce the problem with zero-copy, despite the horrible log formatting. While many of the applied settings are entirely pointless if not outright harmful, this does reproduce the issue.

**Comment by Xaymar** [comment]
So far this happens exactly after the 4th frame, with an unknown problem in libavutil. Most likely will have to pull out more debugging tools to figure this one out.

**Comment by Xaymar** [comment]
After fixing every possible bad behavior on the StreamFX side, and verifying that StreamFX's behavior matches OBS Studio, it still happens. With all the fixes applied, there is now a 1 in 3 chance to encounter either a rendering freeze, a crash, or normal behavior. This appears to be a problem unique to this configuration.

**Comment by Xaymar** [comment]
Appears that this bug can be triggered on command by setting Reference Frames to 0, which is completely invalid anyway. Why NVENC does not immediately exit from this state is unknown, possibly it isn't checking this boundary at all.

**Comment by Xaymar** [comment]
Another deep dive later I have found out more information, though I still have no real clue what is happening to the application state. I did manage to track down the location at which things "break", but not why.

Through disassembly and comparing generated code, I was able to track down the following stack trace:
- av_hw_frame_get_buffer
- d3d11va_get_buffer
- av_buffer_pool_get
- pthread_mutex_lock
- WaitForSingleObject 

For unknown reasons, the handle it waits on is invalid at this point, which hints at a memory corruption much earlier on. There is no indication of _where_ it would occur, but this is so far the location that had the most freezes occur.

**Comment by Xaymar** [comment]
Going to put this on the backburner. I have absolutely no idea what is causing this, as the code appears to match up with what FFmpeg itself shows as an example. I have a few PRs addressing some of the relevant fixes, but none address this issue entirely.

**Comment by LewisTehMinerz** [comment]
> which hints at a memory corruption much earlier on

This appears to be a correct assumption. I tried again today and I was able to catch an event which reports a `STATUS_HEAP_CORRUPTION` exception. Whilst I don't believe it will be much more helpful than you already attaching a debugger, I'll just throw it here for good measure.

<details>
<summary>Eventvwr vomit</summary>

```xml
Log Name:      Application
Source:        Application Error
Date:          26/10/2022 21:05:09
Event ID:      1000
Task Category: Application Crashing Events
Level:         Error
Keywords:      
User:          Lewis-PC\Lewis
Computer:      Lewis-PC
Description:
Faulting application name: obs64.exe, version: 28.0.3.0, time stamp: 0x63389406
Faulting module name: ntdll.dll, version: 10.0.22621.746, time stamp: 0x8a328c67
Exception code: 0xc0000374
Fault offset: 0x000000000010c249
Faulting process ID: 0x0x5968
Faulting application start time: 0x0x1D8E9758DF1A717
Faulting application path: C:\Program Files\obs-studio\bin\64bit\obs64.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report ID: 05ae2b70-48d7-43d7-8012-92545360c515
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" />
    <EventID>1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2022-10-26T20:05:09.7754345Z" />
    <EventRecordID>17390</EventRecordID>
    <Correlation />
    <Execution ProcessID="37124" ThreadID="3984" />
    <Channel>Application</Channel>
    <Computer>Lewis-PC</Computer>
    <Security UserID="S-1-5-21-3865059396-1198155336-3972367085-1001" />
  </System>
  <EventData>
    <Data Name="AppName">obs64.exe</Data>
    <Data Name="AppVersion">28.0.3.0</Data>
    <Data Name="AppTimeStamp">63389406</Data>
    <Data Name="ModuleName">ntdll.dll</Data>
    <Data Name="ModuleVersion">10.0.22621.746</Data>
    <Data Name="ModuleTimeStamp">8a328c67</Data>
    <Data Name="ExceptionCode">c0000374</Data>
    <Data Name="FaultingOffset">000000000010c249</Data>
    <Data Name="ProcessId">0x5968</Data>
    <Data Name="ProcessCreationTime">0x1d8e9758df1a717</Data>
    <Data Name="AppPath">C:\Program Files\obs-studio\bin\64bit\obs64.exe</Data>
    <Data Name="ModulePath">C:\WINDOWS\SYSTEM32\ntdll.dll</Data>
    <Data Name="IntegratorReportId">05ae2b70-48d7-43d7-8012-92545360c515</Data>
    <Data Name="PackageFullName">
    </Data>
    <Data Name="PackageRelativeAppId">
    </Data>
  </EventData>
</Event>
```

</details>

> So far this happens exactly after the 4th frame

For me, I can record/stream randomly between a few seconds and a good few minutes before the crash occurs, and the resulting MKV does actually work fine up until that point. [Here is an example MKV, where it crashed after 53 seconds.](https://discord.coffee/96NFYTr.mkv) Not sure if there'll be anything useful but I'm providing as much as I can that might be of assistance.

I have tried with both GeForce Experience Instant Replay on and off, and I couldn't seem to get a difference. I was able to record 5 minutes of gameplay each time with no issue repeatedly, though I imagine a real 'torture test' for this would be to attempt something like 30 minutes.

**Comment by Xaymar** [comment]
Some people reported that manually patching the FFmpeg binaries within OBS Studio fixes the issue. As my code appears to match the FFmpeg examples, this is probably a possible fix.

**Comment by Xaymar** [comment]
> For me, I can record/stream randomly between a few seconds and a good few minutes before the crash occurs

This is due to the corruption taking a bit before it hits critical structures. With debug markers and guards set, the corruption can be tracked to happen exactly after 4 frames.

**Comment by Xaymar** [comment]
Well, since the bug isn't from StreamFX as far as i can tell, the only fix seems to be to update FFmpeg binaries.

**Comment by LewisTehMinerz** [comment]
I followed the guide for updating FFmpeg (although changing the instructions from -58 to -59, since that's what we're on now for OBS 28) and it seems to be stable. It does indeed appear that this is a bug in OBS' FFmpeg. I haven't yet tested it extensively, but I've been going for a lot longer than I had been during my previous attempts. I will attempt to do a regular stream or recording later and return with my results.

**Comment by Canowyrms** [comment]
Came here with the same issue. Followed this thread, and followed the upgrade guide ([linked here](https://github.com/Xaymar/obs-StreamFX/wiki/Guide-OBS-Studio-FFmpeg-Upgrade)). So far, so good - I've had OBS w/ Replay Buffer running for a couple hours straight with no crashes.

**Comment by MilanPutnik** [comment]
Following up on the issue at hand - after updating OBS to the latest v28.1.1 as well as the whole StreamFX suite v0.12.0 to the latest alpha 151 yesterday, I've successfully finished a comprehensive test with an hour long recording and streaming.

(OBS 18.1.1 default FFpmeg binaries)

I've pushed all machines together with the streaming machine and both the OBS and codec to their limits with a very complex multi-scene profile I've made for my requirements.
It is highly interactive with complex stinger and shader-based transitions and effects, rich use of layer blending and scale filtering, browser and NDI video and VBAN audio sources of both local and remote LAN as well as encrypted WAN origin, automated events running on various animated elements using my custom scripts and macros trough Advanced Scene Switcher....and a literal shitload of all sorts of other stuff I use in my lab across 3 workstations I dedicated solely for the purpose of streaming/broadcasting and real-time production of...my thing...that I have running on 2 other workstations.

That is 5 machines in total all taking part via 25Gbit/s networking interconnect through nVidia mellanox ConnectX-6 accelerated NICs.

Windows 11 22H2 ecosystem 100%
nVidia Maxwell, Turing and Ampere based GPU's
Intel Sky Lake, Raptor Lake + AMD Zen 3 CPU's 
NDI framework, OBS-NDI and all other OBS addons updated to their latest respectable versions as of yesterday (11/04/2022)

**If it survived 1h of everything I've done to it yesterday, at least this particular issue seems to finally be solved at present.**
**I'll give it another test go just in case.**
